internal/provider/archetype_data_source.go

```diff
@@ -689,23 +689,38 @@ func policyAssignmentType2ArmPolicyAssignment(pamap map[string]PolicyAssignmentT
 		dst.Type = to.Ptr(policyAssignementType)
 		dst.Properties.DisplayName = to.Ptr(src.DisplayName.ValueString())
 
+		policyDefinitionMode := "NotSupported"
+
 		// Set policy definition id.
 		if isKnown(src.PolicyDefinitionName) {
 			n := src.PolicyDefinitionName.ValueString()
 			if !az.PolicyDefinitionExists(n) {
 				return nil, fmt.Errorf("policy definition %s not found in AlzLib", n)
 			}
 			dst.Properties.PolicyDefinitionID = to.Ptr(fmt.Sprintf(policyDefinitionIdFmt, n))
+			mode, err := az.GetPolicyDefinitionMode(n)
+			if err != nil {
+				return nil, fmt.Errorf("unable to get policy definition mode for policy definition %s: %w", n, err)
+			}
+			policyDefinitionMode = mode
+
 		}
 		if isKnown(src.PolicySetDefinitionName) {
 			n := src.PolicySetDefinitionName.ValueString()
 			if !az.PolicySetDefinitionExists(n) {
 				return nil, fmt.Errorf("policy set definition %s not found in AlzLib", n)
 			}
 			dst.Properties.PolicyDefinitionID = to.Ptr(fmt.Sprintf(policySetDefinitionIdFmt, n))
+			policyDefinitionMode = "PolicySet"
 		}
 		if isKnown(src.PolicyDefinitionId) {
```

Comment:
This scenario covers assigning a built-in policy or policy set.

Therefore we need to detect this and handle both cases.

At the moment we are assuming this is not a policy set, though this assumption could be incorrect.